lib/dtutils/processor.lua

```diff
@@ -0,0 +1,266 @@
+
+    local result = dp.extract_collection_path(image_table)
+      image_table - table - a table of images such as supplied by the exporter or by libPlugin.build_image_table
+
+  DESCRIPTION
+    extract_collection_path looks at the first image in the image_table and returns the path
+
+  RETURN VALUE
+    result - string - the collection path on success, or nil if there was an error
+
+  LIMITATIONS
+    The collection path is determined from the first image.  If the table consists of images from different
+    collections, then only the first collection is used.
+
+]]
+
+dtutils_processor.libdoc.functions[#dtutils_processor.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Return_Value", "Limitations"},
+  Name = [[extract_collection_path - extract the collection path from an image table]],
+  Synopsis = [[local dp = require "lib/dtutils.processor"
+
+    local result = dp.extract_collection_path(image_table)
+      image_table - table - a table of images such as supplied by the exporter or by libPlugin.build_image_table]],
+  Description = [[extract_collection_path looks at the first image in the image_table and returns the path]],
+  Return_Value = [[result - string - the collection path on success, or nil if there was an error]],
+  Limitations = [[The collection path is determined from the first image.  If the table consists of images from different
+    collections, then only the first collection is used.]],
+}
+
+function dtutils_processor.extract_collection_path(image_table)
+  collection_path = nil
+  for i,_ in pairs(image_table) do
+    collection_path = i.path
+    break
+  end
+  return collection_path
+end
+
+--[[
+  NAME
+    make_output_filename- make an output filename from an image list
+
+  SYNOPSIS
+    local dp = require "lib/dtutils.processor"
+    
+    local result = dtutils.make_output_filename(img_list)
+      img_list - string - a space separated list of filenames
+
+  DESCRIPTION
+    make_output_filename takes a string of filenames, breaks them apart, then
+    combines them in a way that makes some sense.  This is useful in routines
+    that do panoramas, hdr, or focus stacks.  The returned filename is a representation
+    of the files used to construct the image.  If there are 3 or fewer images, then the file
+    basenames are concatenated with a separator. If there is more than 3 images, the first and 
+    last file basenames are concatenated with a separator.
+
+  RETURN VALUE
+    result - string - the constructed filename on success, nil on error
+
+]]
+
+dtutils_processor.libdoc.functions[#dtutils_processor.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Return_Value"},
+  Name = [[make_output_filename- make an output filename from an image list]],
+  Synopsis = [[local dp = require "lib/dtutils.processor"
+    
+    local result = dtutils.make_output_filename(img_list)
+      img_list - string - a space separated list of filenames]],
```

Comment:
wouldn't an image table be more consistant with the other functions in that block, rather that a space-separated list of filenames ?